core/src/main/scala/kafka/utils/ZkUtils.scala

```diff
@@ -707,6 +714,14 @@ class ZkUtils(val zkClient: ZkClient,
     }
   }
 
+  def getTopicPartitionCount(topic: String, defaultNumPartitions: Int = 50): Int = {
+    val topicData = getPartitionAssignmentForTopics(Seq(topic))
+    if (topicData(topic).nonEmpty)
```

Comment:
hmm, it is a bit tricky with `getOrElse` since you need to call `size()` afterwards right?